web/packages/teleport/src/Roles/RoleEditor/StandardEditor/MetadataSection.tsx

```diff
@@ -16,70 +16,156 @@
+          if (isEditing || name === '') {
+            return;
+          }
+          try {
+            await resourceService.fetchRole(name);
+            // `fetchRole` will throw an error if HTTP/404 is returned,
+            // indicating that there's no such role. If we end up here, it
+            // means that there is already a role with this name, so we need to
+            // report it back to the model.
+            //
+            // Compatibility note: if we hit a proxy that doesn't yet support
+            // this endpoint, we just get a 404 anyway, which simply means the
+            // validation won't work in this case, but it won't be a
+            // catastrophic failure; the user will just get their role rejected
+            // by the server and will see a server error instead.
+            dispatch({
+              type: ActionType.SetRoleNameCollision,
+              payload: true,
+            });
+          } catch (e) {
+            // If there's no such role, don't do anything. That's expected.
+            // Otherwise, print the exception, since it would be swallowed by
+            // `useAsync` anyway, and we don't use the returned attempt object.
+            if (!(e instanceof ApiError && e.response.status === 404)) {
+              console.error(e);
+            }
+          }
+        },
+        [resourceService, dispatch]
+      )
+    );
+
+    // Getting callback caching right is especially important here, as we want
+    // always to use the correct version of `checkForNameCollisionNow`, but we
+    // also don't want to call `debounce` on every render, as it would defeat
+    // the purpose of debouncing altogether.
+    const checkForNameCollision = useCallback(
+      debounce(checkForNameCollisionNow, 500),
+      [checkForNameCollisionNow]
+    );
+
+    useEffect(() => {
+      // Check if the default name is available right after the component is
+      // rendered for the first time.
+      checkForNameCollision(value.name);
+    }, []);
+
+    function handleChange(newValue: MetadataModel) {
+      dispatch({ type: ActionType.SetMetadata, payload: newValue });
+    }
+
+    function handleNameChange(e: React.ChangeEvent<HTMLInputElement>) {
+      const name = e.target.value;
+      handleChange({ ...value, name, nameCollision: false });
+      checkForNameCollision(name);
+    }
+
+    return (
+      <Flex flexDirection="column" gap={3}>
+        <SectionPadding>Basic information about this role</SectionPadding>
+        <Box
+          border={1}
+          borderColor={theme.colors.interactive.tonal.neutral[0]}
+          borderRadius={3}
+          p={3}
+        >
+          <FieldInput
+            label="Role Name"
+            required
+            placeholder="Enter Role Name"
+            value={value.name}
+            disabled={isProcessing}
+            readonly={isEditing}
+            rule={requiredAll(
+              precomputed(validation.fields.name),
```

Comment:
Cool JSDoc for `precomputed`, I was wondering what it is and was able to answer that question for myself with a simple hover.